pkg/genlib/fields/load.go

```diff
@@ -61,42 +62,99 @@ func makePackageURL(baseURL, integration, version string) (*url.URL, error) {
 	return u, nil
 }
 
-func getFieldsFiles(ctx context.Context, packageURL *url.URL, dataStream string) ([]byte, error) {
-	body, err := getFromURL(ctx, packageURL.String())
+func makeDownloadURL(baseURL, donwloadPath string) (*url.URL, error) {
+
+	u, err := url.Parse(baseURL)
 	if err != nil {
 		return nil, err
 	}
 
-	var assetsPayload struct {
-		Assets []string `json:"assets"`
+	u.Path = path.Join(u.Path, donwloadPath)
+	return u, nil
+}
+
+func getFieldsFiles(ctx context.Context, baseURL, integration, dataStream, version string) ([]byte, error) {
+	packageURL, err := makePackageURL(baseURL, integration, version)
+	if err != nil {
+		return nil, err
 	}
 
-	if err = json.Unmarshal(body, &assetsPayload); err != nil {
+	r, err := getFromURL(ctx, packageURL.String())
+	if err != nil {
 		return nil, err
 	}
 
-	fieldsFilesURL := make([]string, 0)
-	prefixFieldsPath := path.Join(packageURL.Path, dataStreamSlug, dataStream, fieldsSlug)
-	for _, assetPath := range assetsPayload.Assets {
-		if !strings.HasPrefix(assetPath, prefixFieldsPath) {
-			continue
-		}
+	var downloadPayload struct {
+		Download string `json:"download"`
+	}
 
-		fieldsFilesURL = append(fieldsFilesURL, assetPath)
+	body, err := ioutil.ReadAll(r)
+	if err = json.Unmarshal(body, &downloadPayload); err != nil {
+		return nil, err
 	}
 
+	downloadURL, err := makeDownloadURL(baseURL, downloadPayload.Download)
+	r, err = getFromURL(ctx, downloadURL.String())
+	if err != nil {
+		return nil, err
+	}
+
+	h := sha256.New()
+	h.Write([]byte(downloadURL.String()))
+	prefix := hex.EncodeToString(h.Sum(nil))
+
+	packageTempDir, err := os.MkdirTemp(settings.CacheDir(), prefix)
+	if err != nil {
+		return nil, err
+	}
+	packageArchive := path.Join(packageTempDir, "package.zip")
+	f, err := os.Create(packageArchive)
+	if err != nil {
+		return nil, err
+	}
+
+	_, err = io.Copy(f, r)
+	_ = f.Close()
+	if err != nil {
+
+		return nil, err
+	}
+
+	archive, err := zip.OpenReader(packageArchive)
+	if err != nil {
+		return nil, err
+	}
+	defer func() {
+		_ = archive.Close()
+	}()
+
+	prefixFieldsPath := path.Join(fmt.Sprintf("%s-%s", integration, version), dataStreamSlug, dataStream, fieldsSlug)
+
 	var fieldsContent string
-	for _, fieldsFileURL := range fieldsFilesURL {
-		packageURL.Path = fieldsFileURL
+	for _, z := range archive.File {
+		if z.FileInfo().IsDir() {
+			continue
+		}
+
+		if !strings.HasPrefix(z.Name, prefixFieldsPath) {
+			continue
+		}
 
-		body, err := getFromURL(ctx, packageURL.String())
+		fieldsFileName := z.Name
+		zr, err := z.Open()
 		if err != nil {
 			return nil, err
 		}
 
-		key := strings.TrimSuffix(filepath.Base(fieldsFileURL), filepath.Ext(fieldsFileURL))
+		fieldsFileContent, err := ioutil.ReadAll(zr)
+		if err != nil {
+			return nil, err
+		}
+
+		_ = zr.Close()
```

Comment:
Why ignore error on Close()?

Reply:
once we extracted the fields we don't really care if we cannot close the zip reader, do we? :)